StackableRunes/mod.js

```diff
@@ -137,26 +137,59 @@ for (let i = 0; i < ITEM_TYPES.length; i = i + 1) {
 
 // this is behind a config option because it's *a lot* of recipes...
 if (config.convertWhenDestacking) {
-  for (let i = 0; i < ITEM_TYPES.length; i = i + 1) {
-    const itemtype = ITEM_TYPES[i];
-    const stacktype = converItemTypeToStackItemType(itemtype);
-    for (let j = 2; j <= 500; j = j + 1) {
+  for (let i = 0; i < ITEM_TYPES.length - 1; i = i + 1) {
+    const itemtype_cur = ITEM_TYPES[i];
+    const stacktype_cur = converItemTypeToStackItemType(itemtype_cur);
+    const itemtype_up = ITEM_TYPES[i + 1];
+    const stacktype_up = converItemTypeToStackItemType(itemtype_up);
+    for (let j = 2; j <= config.maxInStack; j = j + 1) {
       cubemain.rows.push({
-        description: `Stack of ${j} ${itemtype} -> Stack of ${
-          j - 1
-        } and Stack of 1`,
+        description: `Stack of ${j} ${stacktype_cur} -> Stack of ${j - 1} ${stacktype_cur} and ${itemtype_cur}`,
         enabled: 1,
         version: 0,
         op: 18, // skip recipe if item's Stat.Accr(param) != value
         param: 70, // quantity (itemstatcost.txt)
         value: j, // only execute rule if quantity == j
         numinputs: 1,
-        'input 1': stacktype,
-        output: `"${stacktype},qty=${j - 1}"`,
-        'output b': `"${itemtype},qty=1"`,
+        'input 1': stacktype_cur,
+        output: `"${stacktype_cur},qty=${j - 1}"`,
+        'output b': `"${itemtype_cur},qty=1"`,
         '*eol': 0,
       });
     }
+    if (config.upgradeStack) {      
+      cubemain.rows.push({
+        description: `Stack of 1 ${stacktype_cur} and ${itemtype_cur} -> Stack of 1 ${stacktype_up}`,
+        enabled: 1,
+        version: 0,
+        op: 18, // skip recipe if item's Stat.Accr(param) != value
+        param: 70, // quantity (itemstatcost.txt)
+        value: 1, // only execute rule if quantity == j
+        numinputs: 2,
+        'input 1': stacktype_cur,
+        'input 2': itemtype_cur,
+        'output': `"${stacktype_up},qty=1"`,
+        '*eol': 0,
+      });
+      for (let j = 2; j <= config.maxInStack; j = j + 1) {
+        cubemain.rows.push({
+          description: `Stack of ${j} ${stacktype_cur} and ${itemtype_cur} -> Stack of ${
+            j - 1
+          } ${stacktype_cur} and Stack of 1 ${stacktype_up}`,
+          enabled: 1,
+          version: 0,
+          op: 18, // skip recipe if item's Stat.Accr(param) != value
+          param: 70, // quantity (itemstatcost.txt)
+          value: j, // only execute rule if quantity == j
+          numinputs: 2,
+          'input 1': stacktype_cur,
+          'input 2': itemtype_cur,
+          output: `"${stacktype_cur},qty=${j - 1}"`,
+          'output b': `"${stacktype_up},qty=1"`,
+          '*eol': 0,
+        });
```

Comment:
This option should serve to simplify recipes for upgrading runes from the stack (only two clicks). For LR, MR the difference in price (2 runes instead of 3) is not fundamental.
If use recipes similar to the original, then the whole point is lost due to stacking / destacking and additional ingredients.

I just suggested a homemade improvement. Whether or not to add is your decision.

Thanks for the mods and GL HF!